File: LeetCode/May challenge/Day 21/Day_21.py
class Solution:
    def countSquares(self, matrix: List[List[int]]) -> int:
        ans, row, col = 0, len(matrix), len(matrix[0])
        for i in range(col):
            ans += matrix[row - 1][i]
        print(ans)
        for i in range(row):
            ans += matrix[i][col - 1]
        print(ans)
        ans = ans - matrix[row - 1][col - 1]
        print(ans)
        for i in range(row - 2, -1, -1):
            for j in range(col - 2, -1, -1):
                if matrix[i][j] == 1:
                    matrix[i][j] = 1 + min(matrix[i + 1][j + 1], matrix[i][j + 1], matrix[i + 1][j])
                else:
                    matrix[i][j] = 0
                ans = ans + matrix[i][j]
        return ans

File: LeetCode/May challenge/Day 21/Day_21.cpp
class Solution {
public:
    int countSquares(vector<vector<int>>& matrix) {
        int sum=0;
        for(int i=0;i<matrix.size();i++)
        {
            for(int j=0;j<matrix[0].size();j++)
            {
                if(i && j && matrix[i][j]!=0)
                    matrix[i][j]=min({matrix[i-1][j],matrix[i][j-1],matrix[i-1][j-1]})+1;
                sum+=matrix[i][j];
            }
        }
        return sum;
    }
};
